web/src/engine/websites/ReaperScans.ts

```diff
@@ -48,66 +41,44 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override ValidateMangaURL(url: string): boolean {
-        return new RegExpSafe(`^${this.URI.origin}/comics/[^/]+$`).test(url);
+        return new RegExpSafe(`^${this.URI.origin}/series/[^/]+$`).test(url);
     }
 
     public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
-        return this.interactionTaskPool.Add(async () => Common.FetchMangaCSS.call(this, provider, url, 'h1.text-xl'), Priority.Normal);
+        const slug = url.match(/\/series\/([^/]+)/)[1];
+        const manga = await this.interactionTaskPool.Add(async () => FetchJSON<APIManga>(new Request(new URL(`/series/${slug}`, this.apiUrl))), Priority.Normal);
+        return new Manga(this, provider, manga.id.toString(), manga.title.trim());
     }
 
     public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
         const mangaList = [];
         for (let page = 1, run = true; run; page += 1) {
-            const mangas = await this.interactionTaskPool.Add(async () => Common.FetchMangasSinglePageCSS.call(this, provider, `/comics?page=${page}`, 'div.grid li a.text-sm'), Priority.Normal);
+            const mangas = await this.interactionTaskPool.Add(async () => this.GetMangasFromPage(page, provider), Priority.Normal);
             mangas.length > 0 ? mangaList.push(...mangas) : run = false;
         }
         return mangaList;
     }
 
-    private ExtractChapters(manga: Manga, node: ParentNode): Chapter[] {
-        const elements = [ ...node.querySelectorAll<HTMLAnchorElement>('ul li a') ];
-        return elements.map(element => new Chapter(this, manga, element.pathname, element.querySelector('div.truncate p.truncate').textContent.trim()));
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(new Request(new URL(`/query?page=${page}&perPage=100&series_type=Comic&adult=true`, this.apiUrl)));
```

Comment:
Too much parameters make the url unreadable, consider composing an URL and set parameters individually.
```typescript
FetchMangas(...) {
const uri = new URL('...', this.apiURL);
uri.searchParams.set('perPage', '100');
uri.searchParams.set('series_type', 'Comic');
for(int page; ...; page++) {
    uri.searchParams.set('page', page.toString());
    const mangas = await FetchMangasFromPage(uri, ...);
}

FetchMangasFromPage(uri, ...) {}
```